app/systemd/entry.sh

```diff
@@ -1,14 +1,65 @@
 #!/bin/bash
 
- set -m
+set -m
 
- # Send SIGTERM to child processes of PID 1.
+# This command only works in privileged container
+ip link add dummy0 type dummy &> /dev/null
+if [[ $? == 0 ]]; then
+	PRIVILEGED=true
+	# clean the dummy0 link
+	ip link delete dummy0 &> /dev/null
```

Comment:
It might be cleaner to also trap certain signals to ensure this link is cleaned up no matter what.

Reply:
@xginn8 I think it's a bit tricky, do you expect any race condition? We are setting up traps for elsewhere as well.

How about something like this (that cleans up before trying, just in case?)
```
## Privileged container check
function dummy0_remove()
{
	# clean the dummy0 link
	ip link delete dummy0 &> /dev/null || true
}
# This command only works in privileged container
dummy0_remove
if ip link add dummy0 type dummy &> /dev/null; then
	dummy0_remove
	PRIVILEGED=true
else
	PRIVILEGED=false
fi
```

Reply:
yep, i think this is a bit better!